mozilla-mobile/firefox-ios
# Add FXIOS-10163 [Homepage] Initial redux implementation

## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10163)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/22265)

## :bulb: Description
* Add initial redux foundation for homepage
* Created main `HomepageState` for the parent state
* Created example substate `HeaderState` for the child section states
* Add two type of actions `initialize` and `updateHeader` for demo purposes in how we can dispatch an action and have that update the collection view

Overall, added code to demonstrate an action taken and how the header section updates based on that action. 
## :pencil: Checklist
You have to check all boxes before merging

## Testing Steps
* Turn on the new homepage with feature flags
* View the new homepage by adding a new tab
* Confirm that the initial sections are loaded
* Tap on either "First" cell and view the changes in the sections 

https://github.com/user-attachments/assets/0ee9b135-ac13-4a5b-a204-bd31170fabf2
